packages/backend/src/ee/services/ai/types/aiAgent.ts

```diff
@@ -0,0 +1,30 @@
+import { AiChatMessage } from '@lightdash/common';
+import {
+    GetExploreFn,
+    GetPromptFn,
+    RunMiniMetricQueryFn,
+    SearchFieldsFn,
+    SendFileFn,
+    UpdateProgressFn,
+    UpdatePromptFn,
+} from './aiAgentDependencies';
+import { AiAgentExploreSummary } from './aiAgentExploreSummary';
+
+export type AiAgentArgs = {
+    openaiApiKey: string;
+    promptUuid: string;
+    agentName: string;
+    instruction: string | null;
+    messageHistory: AiChatMessage[];
+    aiAgentExploreSummaries: AiAgentExploreSummary[];
+    maxLimit: number;
+};
+export type AiAgentDependencies = {
+    getExplore: GetExploreFn;
+    searchFields: SearchFieldsFn | undefined;
+    runMiniMetricQuery: RunMiniMetricQueryFn;
+    getPrompt: GetPromptFn;
+    sendFile: SendFileFn;
```

Comment:
(outside of the scope of this PR) It feels a bit wrong for me that tool calls can send files, I think anything that reads more data is okay from the tool calls but any output should be declaratively stated and only executed on after the whole agent loop ends. Might be wrong on this but that's my intuition, wdyt?

Reply:
I agree. workflow has a lot of side effects. we can iterate on this, at this point I'm just replicating what we had before 1:1 with a bit of type cleanups